cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/runtimejob/DefaultRuntimeJob.java

```diff
@@ -499,6 +529,26 @@ List<Module> createModules(RuntimeJobEnvironment runtimeJobEnv, CConfiguration c
     modules.add(new AuthenticationContextModules().getProgramContainerModule(cConf));
     modules.add(new MetricsClientRuntimeModule().getDistributedModules());
     modules.add(new MessagingServerRuntimeModule().getStandaloneModules());
+    modules.add(new NamespaceQueryAdminModule());
```

Comment:
Have you considered unifying it to use the `DistributedProgramContainerModule`?

Reply:
will take a look at this